### ISSUE TITLE
Clear execution times w/"Restart & Clear Output"

### ISSUE THREAD

**Issue by jakirkham** [ipython-contrib/jupyter_contrib_nbextensions#1132]
Not sure if there is a way to do this, but it would be great if selecting in the Notebook "Kernel" > "Restart & Clear Output" would also clear the execution times. Am aware that this can be done with "Cell" > "Execution Timings" > "Clear (all)". Just sometimes forget to do the latter when doing the former. :)

**Comment by jcb91** [comment]
mm, yeah, I guess we could patch the action `jupyter-notebook:confirm-restart-kernel-and-clear-output` to also clear ExecuteTime data, that would be more convenient, I guess? However, given the mentality that the execute timings should be cleared along with outputs, perhaps it would be better to patch the cell's clear_output method, so that they get removed whenever output is cleared (e.g. through `Cell > current outputs > clear` menu item). Thoughts?

**Comment by jakirkham** [comment]
If there was a way to register for some sort of cell event, maybe that would give us an opportunity to react to this clear event and do the cleanup. Though I have no idea if this is possible. I think you know way more in this area than I do. :)

**Comment by jcb91** [comment]
Haha actually, it seems I should have checked the readme, as I've already written options for doing exactly this - see [the options section in the readme](https://github.com/ipython-contrib/jupyter_contrib_nbextensions/blob/f105d1db7af86092915907f63e3b8af28f8e47e1/src/jupyter_contrib_nbextensions/nbextensions/execute_time/readme.md#options), which details the options `ExecuteTime.clear_timings_on_clear_output` and `ExecuteTime.clear_timings_on_kernel_restart`, both of which are off (`false`) by default, but should give you what you want, hopefully. As the readme mentions, the easiest way to set them is with the [jupyter_nbextensions_configurator](https://github.com/Jupyter-contrib/jupyter_nbextensions_configurator), which gets installed as a dependency of this repo, but you can also go edit the config more directly if you'd rather. Let me know if you're after something subtly different to what's already implemented, or if it doesn't do what it claims to, of course :)

**Comment by jakirkham** [comment]
Oops. 😳 Sorry that's my bad. I did see the readme, but totally missed that section. Let me give that a try. Thanks for pointing it out and adding this feature. 😄

**Comment by jakirkham** [comment]
So I just gave this a try. Seems that one needs to close any existing notebooks and reopen them for the setting to take effect. Once that is done, it works perfectly. Thanks again. :)

**Comment by jcb91** [comment]
> Seems that one needs to close any existing notebooks and reopen them for the setting to take effect

You'd need to refresh the page, as config settings are only loaded when the nbextension loads. Is that what you meant by closing & reopening? You didn't need to restart servers or anything, I hope?

**Comment by jakirkham** [comment]
Ah ok. Will keep that in mind.

No nothing like closing the server. Had a test notebook I was trying this with. So just closed that file and reopened it (shutting down its kernel in the process). Sounds like the kernel could have stayed running too though.

**Comment by jcb91** [comment]
> Sounds like the kernel could have stayed running too though

Yes, it should be fine to just refresh the page, leaving kernel as-is

**Comment by jakirkham** [comment]
Not sure if you would like this in a new issue. Happy to move if needed.

Was wondering if there is a way to set the [config options]( https://github.com/ipython-contrib/jupyter_contrib_nbextensions/blob/f105d1db7af86092915907f63e3b8af28f8e47e1/src/jupyter_contrib_nbextensions/nbextensions/execute_time/readme.md#options ) in the sys-prefix location instead of the home directory. Tried playing with this, but didn't figure it out. Guessing there is something obvious I'm overlooking. Any suggestions appreciated. 😄

**Comment by jcb91** [comment]
> Guessing there is something obvious I'm overlooking

Probably just that the configurator uses the notebook server's config API, which exposes only as read-from-all write-to-user.

**Comment by jcb91** [comment]
To get round this, we'd either need to add an option to notebook along the lines of "sys-prefix beats user", or get the configurator to expose its own config-style API endpoint supporting such an access mode

**Comment by jcb91** [comment]
Or, perhaps another serverextension which could patch the config manager :thinking:

**Comment by jcb91** [comment]
Oh, and in case it wasn't clear, it's certainly possible to have config in the sys-prefix directory, written by python or whatever other language you feel like to directly edit the json. The restrictions apply only to using the config API

**Comment by jakirkham** [comment]
Sure. Was mainly looking for an API. May hack something together in the interim.

Do you think this is within scope here or should we move it somewhere else (e.g. the Jupyter Notebook repo)?

**Comment by jcb91** [comment]
> Do you think this is within scope here or should we move it somewhere else (e.g. the Jupyter Notebook repo)?

I guess a modification or option for the configurator would make sense on the [configurator repo](https://github.com/Jupyter-contrib/jupyter_nbextensions_comfigurator). For a new, separate serverextension, I'd maybe stick it in its own repo, or possibly here. For modifications to notebook itself, obviously there ;)
